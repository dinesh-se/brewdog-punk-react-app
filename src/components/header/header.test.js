import React from 'react';
import { shallow } from 'enzyme';
import Header from './header';

test('should render Header correctly', () => {
  const appInfo = {
    title: 'Brew Dog',
    subtitle: 'Beer application'
  };
  const wrapper = shallow(<Header title={appInfo.title} subtitle={appInfo.subtitle} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.app-title').text()).toBe(appInfo.title);
  expect(wrapper.find('.app-subtitle').text()).toBe(appInfo.subtitle);
});

test('should render Header with title only', () => {
  const appInfo = {
    title: 'Brew Dog'
  };
  const wrapper = shallow(<Header title={appInfo.title} subtitle={appInfo.subtitle} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.app-title').text()).toBe(appInfo.title);
  expect(wrapper.find('.app-subtitle')).toHaveLength(0);
});