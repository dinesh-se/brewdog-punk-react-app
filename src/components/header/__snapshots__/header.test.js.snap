// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h2
    className="app-title"
  >
    Brew Dog
  </h2>
  <p
    className="app-subtitle"
  >
    Beer application
  </p>
</header>
`;

exports[`should render Header with title only 1`] = `
<header>
  <h2
    className="app-title"
  >
    Brew Dog
  </h2>
</header>
`;
